javatests/arcs/core/storage/handle/CollectionIntegrationTest.kt

```diff
@@ -93,6 +97,19 @@ class CollectionIntegrationTest {
         assertThat(collectionB.fetchAll()).containsExactly(person.toRawEntity())
     }
 
+    @Test
+    fun addingElementsToA_showsUpInQueryOnB() = runBlockingTest {
+        val miles = Person("Miles", 55, true, emptySet())
+        val jason = Person("Jason", 35, false, setOf("Watson"))
+
+        collectionA.store(miles.toRawEntity())
+        collectionA.store(jason.toRawEntity())
+
+        assertThat(collectionA.fetchAll()).containsExactly(miles.toRawEntity(), jason.toRawEntity())
+        assertThat(collectionB.query(55)).containsExactly(miles.toRawEntity())
+        assertThat(collectionB.query(35)).containsExactly(jason.toRawEntity())
+    }
```

Comment:
It'd be worth adding a test case for when no elements matching the query, and another for when you call query and there is no refinement